tests/unit/test_stream_app.py

```diff
@@ -615,3 +615,42 @@ def stream_app(event, api, cache):
     result_event: StreamDepthEvent = stream_app(event, context)[0]
 
     assert result_event.log_identifier == 'log_identifier'
+
+
+def test_raw_stream_event_with_none_data_field_returns_expected_result(context):
+    """Make sure that raw stream event with empty data field
+    can be handled without validation exception.
+    """
+
+    @stream
+    def stream_app(event, api, cache):
+        pytest.fail(
+            "Stream app call should be skipped "
+            "because there is no data to build an event"
+        )
+
+    event = [
+        {
+            "metadata": {
+                "app_stream_id": 123,
+                "apps": {"test-provider.test-app-name": {"app_connection_id": 456}},
+                "log_type": "time",
+                "source_type": "drilling",
+            },
+            "records": [
+                {
+                    "app": "corva.wits-historical-import",
+                    "asset_id": 1,
+                    "collection": "wits.completed",
+                    "company_id": 80,
+                    "data": None,
+                    "provider": "corva",
+                    "timestamp": 1688999883,
+                    "version": 1,
+                }  # DEVC-627. This record should be filtered out because data is None.
+            ],
+        }
+    ]
+
+    _ = stream_app(event, context)[0]
+    assert True, "App call should be skipped"
```

Comment:
This can never fail:)

Reply:
I agree. But if we remove it will hurt the readability because you expect to see asserts at the end of the test. But I can remove it if needed.